scrollIntoView.js

```diff
@@ -11,7 +11,18 @@ const alignBottom = (scrollingElement, element, offsetBottom) => {
 const isFullVisible = (scrollElement, element, offsetTop, offsetBottom) => {
   const scrollElementRect = scrollElement.getBoundingClientRect();
   const elementRect = element.getBoundingClientRect();
-  if (elementRect.top < (scrollElementRect.top + offsetTop) || elementRect.bottom > (scrollElementRect.bottom - offsetBottom)) {
+
+  let scrollElementTop = scrollElementRect.top;
```

Comment:
@shakil-mansuri-dream this logic seems to have a performance issue, when document is scrolling element, we don't need to get `boundingClientRect` of the `scrollElement`. But, we are getting